File: backend/app/schemas/document.py
from pydantic import BaseModel
from datetime import datetime


class DocumentCreate(BaseModel):
    name: str
    description: str | None = None
    category: str | None = None  # plan, report, contract, other
    file_path: str
    file_type: str | None = None
    file_size: int | None = None


class DocumentUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    category: str | None = None
    version: int | None = None


class DocumentResponse(BaseModel):
    id: int
    folio: str
    name: str
    description: str | None
    category: str | None
    file_path: str
    file_type: str | None
    file_size: int | None
    version: int
    project_id: int
    uploaded_by_id: int | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/document.ts
/**
 * Document type definitions.
 * Aligned with backend schemas: DocumentResponse.
 */

/** Matches backend DocumentResponse schema */
export interface Document {
  id: number;
  folio: string;
  name: string;
  description: string | null;
  category: string | null;
  file_path: string | null;
  file_type: string | null;
  file_size: number | null;
  version: number;
  project_id: number;
  uploaded_by_id: number | null;
  created_at: string;
}

/** Cross-project document view used in DocumentsPage (with project info) */
export interface DocWithProject {
  id: number;
  folio: string;
  name: string;
  category: string;
  file_type: string;
  file_size: number;
  project_name: string;
  project_id: number;
  uploaded_by: string;
  created_at: string;
}

/** Response from file upload endpoint */
export interface UploadResponse {
  file_path: string;
  file_name: string;
  file_type: string;
  file_size: number;
}
